digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataImporter/TrafficSignCsvImporter.scala

```diff
@@ -89,20 +235,155 @@ class TrafficSignCsvImporter(roadLinkServiceImpl: RoadLinkService, eventBusImpl:
 
-    val additionalInfo = getPropertyValue(trafficSignAttributes, "additionalInfo").toString
-    val additionalProperty = if(additionalInfo.nonEmpty)
-      Some(SimplePointAssetProperty(infoPublicId, Seq(PropertyValue(additionalInfo))))
-    else
-      None
+    val typeProperty = SimplePointAssetProperty(typePublicId, Seq(PropertyValue(TrafficSignType.applyNewLawCode(getPropertyValue(trafficSignAttributes, "trafficSignType").toString).OTHvalue.toString)))
+
+    val listPublicIds = Seq(infoPublicId, startDatePublicId, endDatePublicId, municipalityPublicId, mainSignTextPublicId, structurePublicId, conditionPublicId, sizePublicId,
+                            heightPublicId, coatingTypePublicId, signMaterialPublicId, locationSpecifierPublicId, terrainCoordinatesXPublicId, terrainCoordinatesYPublicId,
+                            laneTypePublicId, lanePublicId, lifeCyclePublicId, typeOfDamagePublicId, urgencyOfRepairPublicId, lifespanLeftPublicId, oldTrafficCodePublicId,
+                            oppositeSideSignPublicId
+    )
+    val listFieldNames = Seq("additionalInfo", "startDate", "endDate", "municipalityId", "mainSignText", "structure", "condition", "size", "height", "coatingType", "signMaterial",
+                              "locationSpecifier", "lon", "lat", "laneType", "lane", "lifeCycle", "typeOfDamage", "urgencyOfRepair", "lifespanLeft",
+                              "oldTrafficCode", "oppositeSideSign"
+    )
+
+    val propertiesValues = (listPublicIds, listFieldNames).zipped.map{(publicId, fieldName) =>
+      val propertyInfo = getPropertyValueOption(trafficSignAttributes, fieldName)
+      if(propertyInfo.get != null && propertyInfo.nonEmpty)
+        Some(SimplePointAssetProperty(publicId, Seq(PropertyValue(propertyInfo.get.toString))))
+      else
+        None
+    }
+    //not possible to insert suggested signs through csv
+    val suggestBox = Set(Some(SimplePointAssetProperty(suggestBoxPublicId, Seq(PropertyValue("0")))))
 
-    val typeProperty = SimplePointAssetProperty(typePublicId, Seq(PropertyValue(TrafficSignType.applyTRValue(getPropertyValue(trafficSignAttributes, "trafficSignType").toString.toInt).OTHvalue.toString)))
+    (Set(Some(typeProperty), valueProperty) ++ propertiesValues ++ suggestBox ++ generateBasePanelProperties(trafficSignAttributes)).flatten
+  }
 
-    Set(Some(typeProperty), valueProperty, additionalProperty).flatten
+  private def generateBasePanelProperties(trafficSignAttributes: ParsedProperties): Set[Option[SimplePointAssetProperty]] = {
+
+    def getSingleChoiceValue(target: String): Int = {
+      getPropertyValueOption(trafficSignAttributes, target) match {
+        case Some(targetValue) if !isBlank(targetValue.toString) => targetValue.toString.toInt
+        case _ => trafficSignService.getDefaultSingleChoiceValue
+      }
+    }
+    var res: Seq[AdditionalPanel] = Seq()
+    val minAdditionalPanels = 1
+    val maxAdditionalPanels = 3
```

Comment:
minAdditionalPanels and maxAdditionalPanels are declared in verifyData and generateBasePanelProperties functions.
Why not make those as attributes of class?

Reply:
Suggestion like above.

Reply:
suggestion applied